components/signup.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/**
+ * Manage BuddyPress Signups.
+ *
+ * @since 1.5.0
+ */
+class BPCLI_Signup extends BPCLI_Component {
+
+	/**
+	 * Add a signup.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--user-login=<user-login>]
+	 * : User login for the signup. If none is provided, a random one will be used.
+	 *
+	 * [--user-email=<user-email>]
+	 * : User email for the signup. If none is provided, a random one will be used.
+	 *
+	 * [--meta=<meta>]
+	 * : User meta for the signup.
+	 *
+	 * [--silent=<silent>]
+	 * : Silent the signup creation.
+	 * ---
+	 * default: false.
+	 * ---
+	 *
+	 * ## EXAMPLES
+	 *
+	 *   $ wp bp signup add --user-login=test_user --user-email=teste@site.com
+	 *   $ wp bp signup add --user-login=test_user --user-email=teste@site.com --silent=1
+	 */
+	public function add( $args, $assoc_args ) {
```

Comment:
FYI: values are not required because we need them to be optional to use in the `generate` command below.